### PR TITLE
All on one sheet for 3 browsers

### PR DESCRIPTION
**Pull request by y-tst**
Have created 3 classes - one per each browser with all checks inside

### CODE REVIEW

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class GoogleSearchForWikiEdge {
+
+    @Test
+    public void GoogleSearchForWikiIsNotEmpty() throws IOException {
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: - methods starts from small letter
- your test name didn't describe the purpose of test, at least add browser name into name (e.g. `...ForEdge` )

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        final Logger logger = Logger.getGlobal();
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:27]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        System.setProperty("webdriver.edge.driver", "D:\\Tools\\msedgedriver.exe");
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: All three drivers should be placed into project. I don't have D:\\Tools\\msedgedriver.exe on my laptop, so if I will start your tests - they will fail

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        WebDriver driver = new EdgeDriver();
+        driver.manage().window().maximize();
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        driver.get("http://google.com");
+        new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By.name("q")));
+        driver.manage().window().maximize();
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:33]: duplicate line, same as line 31

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+        WebElement googleSearchField = new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By.name("q")));
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:35]: you can combine line 35 and 38, they are almost equals

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+        Assert.assertEquals(driver.getTitle(), "wikipedia - Пошук Google", "Different from the main Wiki page is opened");
+
+        logger.info(String.format("Wiki page title is '%s', it has length: %s symbols", driver.getTitle(), driver.getTitle().length()));
+
+        Assert.assertTrue(driver.getCurrentUrl().contains("wikipedia"), "Incorrect URL for Wiki page");
+
+        logger.info(String.format("Wiki page length has %s symbols", driver.getPageSource().length()));
+
+        WebElement didYouKnowPictire = driver.findElement(By.xpath ("(//div[contains(@id,'main-dyk')]//img)[1]"));
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: PictUre

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        BufferedImage img = ImageIO.read(screen);
+        BufferedImage destination = img.getSubimage(xcordinate, ycordinate, imageWidth, imageHeight);
+        ImageIO.write(destination, "png", screen);
+        FileUtils.copyFile(screen, new File("target/pictures/DidYouKnowPic1.png"));
+
+        WebElement element = driver.findElement(By.xpath ("(//div[contains(@id,'main-dyk')]//img)[2]"));
+        Screenshot secondPictureInDidYuuKnow = new AShot().shootingStrategy(ShootingStrategies.viewportPasting(1000)).takeScreenshot(driver,element);
+        ImageIO.write(secondPictureInDidYuuKnow.getImage(), "jpg", new File("target/pictures/DidYouKnowPic2.jpg"));
+
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:86]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        int xcordinate = point.getX();
+        int ycordinate = point.getY();
+        int imageWidth = didYouKnowPictire.getSize().getWidth();
+        int imageHeight = didYouKnowPictire.getSize().getHeight();
+        BufferedImage img = ImageIO.read(screen);
+        BufferedImage destination = img.getSubimage(xcordinate, ycordinate, imageWidth, imageHeight);
+        ImageIO.write(destination, "png", screen);
+        FileUtils.copyFile(screen, new File("target/pictures/DidYouKnowPic1.png"));
+
+        WebElement element = driver.findElement(By.xpath ("(//div[contains(@id,'main-dyk')]//img)[2]"));
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: you can describe in `logger.info` these two ways of getting screenshots.

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+
+        WebElement wikiSearchField = new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By.name("search")));
+        wikiSearchField.sendKeys("Test Automation");
+        wikiSearchField.submit();
+
+        Assert.assertEquals(driver.getTitle(), "Поиск «Test Automation» — Википедия", "Search result doesn't contain 'Test automation'");
+
+        logger.info(String.format("Wiki search for Test automation page result -  '%s'", driver.getTitle()));
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:94]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+        driver.quit();
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdFirefox.java [hunk 12]
```diff
@@ -0,0 +1,91 @@
+package WebDriverSeleniumTests;
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdFirefox.java:1]: see my comments from Edge area- here is the same picture

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiChrome.java [hunk 13]
```diff
@@ -0,0 +1,90 @@
+package WebDriverSeleniumTests;
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiChrome.java:1]: see my comments from Edge area - here is the same picture

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 1 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: Ups)

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 2 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:27]: removed

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 3 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: Placed all 3 drivers into resources folder

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 4 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: removed

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 5 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:33]: removed

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 6 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:35]: Why so?
In 35 I'm working with field, in 38 - with button

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 7 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: Fixed everywhere)

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 8 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:86]: removed

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 9 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: Added loggers with explanations

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 10 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:94]: +

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 11 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java]: Extremely inattentive)

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdFirefox.java (hunk 12 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdFirefox.java:1]: yep.

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiChrome.java (hunk 13 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiChrome.java:1]: yep.

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+        WebDriver driver = new ChromeDriver(chromeoptions);
+
+        driver.get("http://google.com");
+        new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By.name("q")));
+        driver.manage().window().maximize();
+
+        WebElement googleSearchField = new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By.name("q")));
+        googleSearchField.sendKeys("wikipedia");
+
+        List<WebElement> googleSearchButton = driver.findElements(By.name("btnK"));
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java]: Why you use List for one button? And why findElementS ? As far as I see - there is only 1 search button?

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+        new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//div[contains(@class, 'g') and contains(., 'wikipedia')]")));
+
+        List<WebElement> searchResults = driver.findElements(By.xpath("//div[contains(@class, 'g') and contains(., 'wikipedia')]"));
+        logger.info(String.format("Number of pages found: %s", searchResults.size()));
+
+        Assert.assertTrue(searchResults.size() > 0, "Empty result for 'Wikipedia' search");
+
+        WebElement firstWikiLinkFromGoogleSearch = new WebDriverWait(driver, 5).until(ExpectedConditions.presenceOfElementLocated(By.xpath("//*[@id='rso']//a/h3")));
+        firstWikiLinkFromGoogleSearch.click();
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java]: Here you can have synchronization issues in future, you need to wait for Wikipedia page fully loaded and only after proceed with other actions

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiEdFirefox.java [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    public void GoogleSearchForWikiIsNotEmpty() throws IOException {
+
+        final Logger logger = Logger.getGlobal();
+
+        System.setProperty("webdriver.edge.driver", "src/test/resources/geckodriver.exe");
+
+        WebDriver driver = new FirefoxDriver();
+        driver.manage().window().maximize();
+
```
Comment (olelysieev-clgx) [inline src/test/java/WebDriverSeleniumTests/googleSearchForWikiEdFirefox.java]: empty line

File: src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java (hunk 6 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/GoogleSearchForWikiEdge.java:35]: Waiter removed

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java (hunk 14 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java]: Just tried how the list method works  - findElement could be used, for sure

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiEdFirefox.java (hunk 16 above)
Comment (y-tst) [reply src/test/java/WebDriverSeleniumTests/googleSearchForWikiEdFirefox.java]: yeap, got it also

File: src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java (hunk 14 above)
Comment (olelysieev-clgx) [reply src/test/java/WebDriverSeleniumTests/googleSearchForWikiChrome.java]: In such case - it is ok

**Review by olelysieev-clgx — APPROVED** [review]

**Pull request merged**